### PR TITLE
(Doc+) Link Gateway Settings to Full Restart

### PR DESCRIPTION
**Pull request by stefnestor**
👋🏽 howdy, team! Mini update to cross-link the [Gateway Settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-gateway.html) to the [Full Cluster Restart](https://www.elastic.co/guide/en/elasticsearch/reference/current/restart-cluster.html#restart-cluster-full) docs.

### CODE REVIEW

File: docs/reference/upgrade/disable-shard-alloc.asciidoc [hunk 1]
```diff
@@ -17,3 +17,5 @@ PUT _cluster/settings
 }
 --------------------------------------------------
 // TEST[skip:indexes don't assign]
+
+You may also consider <<modules-gateway,gateway settings>>.
```
Comment (shainaraskas) [inline docs/reference/upgrade/disable-shard-alloc.asciidoc:21]: this sentence doesn't make the connection to gateway settings clear. consider:

```suggestion
As an alternative to disabling replica allocation, you can adjust your <<modules-gateway,gateway settings>> to [XYZ action]. 
```

Comment (stefnestor) [reply docs/reference/upgrade/disable-shard-alloc.asciidoc:21]: > this sentence doesn't make the connection to gateway settings clear. consider:

That's probably ignorance on my part, sorry. This setting would not be in replacement but in addition to. I believe you would also use gateway settings if your start-up times between nodes were going to be a bit delayed compared to normal and/or to reduce initial strain (from allocation kicking off) while doing discovery.

Comment (shainaraskas) [reply docs/reference/upgrade/disable-shard-alloc.asciidoc:21]: what setting config would you expect them to use in a full cluster restart scenario?

Comment (stefnestor) [reply docs/reference/upgrade/disable-shard-alloc.asciidoc:21]: IME when users set one they set all three at once. But I guess if we were going to link to only one, it'd be `gateway.recover_after_data_nodes`.

Comment (shainaraskas) [reply docs/reference/upgrade/disable-shard-alloc.asciidoc:21]: can you edit this sentence to reflect that they also might want to set these things due to the reasons you explained? we can link to the top of the settings page - I just didn't understand the connection between these two topics

Comment (stefnestor) [reply docs/reference/upgrade/disable-shard-alloc.asciidoc]: Updated, cheers.

File: docs/reference/upgrade/disable-shard-alloc.asciidoc [hunk 2]
```diff
@@ -17,3 +17,7 @@ PUT _cluster/settings
 }
 --------------------------------------------------
 // TEST[skip:indexes don't assign]
+
+You may also consider <<modules-gateway,gateway settings>> when restarting
```
Comment (shainaraskas) [inline docs/reference/upgrade/disable-shard-alloc.asciidoc:21]:
```suggestion
You can also consider <<modules-gateway,gateway settings>> when restarting 
```

**Review by shainaraskas — APPROVED** [review]
just one suggestion, then good to go! thank you 🙏 

remember to add the versions this is relevant to (maybe at least 8.14 + 8.15 so it appears in the current docs) as well as the `auto-backport-and-merge` label :)

**Pull request merged**